Sources/FoundationEssentials/Data/Data+Writing.swift

```diff
@@ -160,9 +165,15 @@ private func createTemporaryFile(at destinationPath: String, inPath: PathOrURL,
             
             // The warning diligently tells us we shouldn't be using mktemp() because blindly opening the returned path opens us up to a TOCTOU race. However, in this case, we're being careful by doing O_CREAT|O_EXCL and repeating, just like the implementation of mkstemp.
```

Comment:
Yes, sadly the same issue does apply to `_mktemp_s`, it only avoids a buffer length security concern. There is a way to use other functions to avoid the TOCTOU condition by getting back a handle/fd instead. But, I was trying to maintain semantics here. I'm happy to replace this with something that avoids the TOCTOU.

Reply:
Right, but now we use mktemp_s on windows but mktemp everywhere else...

Reply:
Yes, `mktemp` is deprecated on Windows in favour of `_mktemp_s`. This is a small API spelling level difference, so I don't think its too big of a deal. Semantically, the function is identical. It is similar to Apple's new security attributes for buffer lengths, it just is a vulnerability mitigation technique.

Reply:
Ok, thanks.